server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/http/BasicAuthenticationStrategy.scala

```diff
@@ -19,23 +19,59 @@
 
 package org.apache.james.jmap.http
 
+import eu.timepit.refined.auto._
+import eu.timepit.refined.api.Refined
+import eu.timepit.refined.collection.NonEmpty
 import javax.inject.Inject
+import org.apache.james.core.Username
 import org.apache.james.mailbox.{MailboxManager, MailboxSession}
 import reactor.core.publisher.Mono
-import reactor.core.scala.publisher.SFlux
+import reactor.core.scala.publisher.{SFlux, SMono}
 import reactor.netty.http.server.HttpServerRequest
+import org.apache.james.jmap.http.UserCredentialParse._
+import org.apache.james.user.api.UsersRepository
+
 import scala.compat.java8.StreamConverters._
 
-class BasicAuthenticationStrategy @Inject() (val basicAuthTokenManager: BasicAuthenticationTokenManager,
-                                             val mailboxManager: MailboxManager) extends AuthenticationStrategy {
+final case class UserCredential(username: StringNonNull, password: StringNonNull)
+
+object UserCredentialParse {
+
+  type StringNonNull = String Refined NonEmpty
+
+  private val BASIC_AUTHENTICATION_PREFIX: String = "Basic "
+
+  def parseUserCredentials(token: String): SMono[UserCredential] = {
+    SMono.just(token)
+      .filter(token => token.contains(BASIC_AUTHENTICATION_PREFIX))
+      .flatMap(token => userCredentialExtraction(token))
+  }
+
+  private def userCredentialExtraction(token: String): SMono[UserCredential] = {
+    SMono.just(token)
+      .map(token => token.split(":"))
+      .filter(array => array.length == 2)
+      .map(array => UserCredential(array.apply(0), array.apply(1)))
+      .switchIfEmpty(SMono.raiseError(throw new IllegalArgumentException("Credential's format invalid")))
+  }
+}
+
+
+
+class BasicAuthenticationStrategy @Inject()(val usersRepository: UsersRepository,
+                                            val mailboxManager: MailboxManager) extends AuthenticationStrategy {
 
   override def createMailboxSession(httpRequest: HttpServerRequest): Mono[MailboxSession] = {
     SFlux.fromStream(() => authHeaders(httpRequest).toScala[Stream])
-      .map(basicAuthToken => basicAuthTokenManager.removeBasicAuthenticationPrefix(basicAuthToken))
-      .filter(basicAuthToken => basicAuthTokenManager.isValid(basicAuthToken))
-      .map(basicAuthToken => basicAuthTokenManager.retrieveUserNameFromCredential(basicAuthToken))
-      .map(username => mailboxManager.createSystemSession(username))
+      .flatMap(token => parseUserCredentials(token))
+      .filter(userCredential => isValid(userCredential))
+      .map(userCredential => mailboxManager.createSystemSession(Username.of(userCredential.username)))
+      .switchIfEmpty(SMono.raiseError(throw new IllegalArgumentException("User doesn't exist or if the password is incorrect")))
```

Comment:
This is likely the role of the authenticator: think about someone just giving JWT credentionals, you don't wan't basic authentication to throw.